src/history.py

```
@@ -0,0 +1,98 @@
+        budget_for_chunk = 0
+
+    # Divide recent_channel_messages into a chunked array if  exceeds max_tokens
+    chunk_size = 1000  # Adjust chunk size as needed
+    chunks = []
+    for i in range(0, len(recent_channel_messages), chunk_size):
+        chunks.append(recent_channel_messages[i:i + chunk_size])
+
+    logging.debug(f"Number of chunks: {len(chunks)}")
+    chunked_responses = []
+    for chunk in chunks:
+        chunked_response = await summarizer.get_summary_from_ai("\n".join(prior_messages),
```

Ruff (0.8.2)

`summarizer` may be undefined, or defined from star imports

(F405)